### PR TITLE
Box dimension integration

### PR DESCRIPTION
**Pull request by mrmdrogers**
Integration with the new ML model for box dimension prediction.

:shipit:  @syedmusamah

### CODE REVIEW

File: src/main/java/com/thredup/configuration/AwsConfiguration.java [hunk 1]
```diff
@@ -15,8 +16,25 @@
     @Value("${aws.ml.aws_secret_access_key}")
     private String secretKey;
 
+    @Value("${aws.ml.predict-endpoint}")
+    private String predictEndpoint;
+    @Value("${aws.ml.weight.model}")
```
Comment (syedmusamah) [inline src/main/java/com/thredup/configuration/AwsConfiguration.java]: update `application.yml.sample` to reflect this

File: src/main/java/com/thredup/datatype/DimensionPredictRequestContext.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.thredup.datatype;
+
+import java.util.Map;
+
+public class DimensionPredictRequestContext {
+    private String orderId;
```
Comment (syedmusamah) [inline src/main/java/com/thredup/datatype/DimensionPredictRequestContext.java]: we can remove order ID usage from this app

File: src/main/java/com/thredup/constants/MerchandisingDepartment.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    static {
+        MERCHANDISING_DEPARTMENT_NAME_MAP = new HashMap<>();
+        MERCHANDISING_DEPARTMENT_NAME_MAP.put("boys", "kids_item_count");
+        MERCHANDISING_DEPARTMENT_NAME_MAP.put("girls", "kids_item_count");
+        MERCHANDISING_DEPARTMENT_NAME_MAP.put("handbags", "handbag_item_count");
+        MERCHANDISING_DEPARTMENT_NAME_MAP.put("juniors", "juniors_item_count");
+        MERCHANDISING_DEPARTMENT_NAME_MAP.put("maternity", "maternity_item_count");
+        MERCHANDISING_DEPARTMENT_NAME_MAP.put("plus", "plus_item_count");
+        MERCHANDISING_DEPARTMENT_NAME_MAP.put("women", "women_item_count");
+        MERCHANDISING_DEPARTMENT_NAME_MAP.put("women_shoes", "women_shoe_item_count");
```
Comment (syedmusamah) [inline src/main/java/com/thredup/constants/MerchandisingDepartment.java]: ("women shoes") (with a space),

also, this is missing the "X" merchandising department
``` SQL
select merchandising_department from items where state = 'listed'
group by 1

boys
girls
handbags
juniors
maternity
plus
women
women shoes
X
```

File: src/main/java/com/thredup/callable/PredictWeightCallable.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public PredictWeightCallable(PredictWeightContext predictWeightContext) {
+        this.predictWeightContext = predictWeightContext;
+    }
+
+    @Override
+    public PredictWeightResponse call() {
+        PredictResult predictResult = null;
+        try {
+            long start = System.currentTimeMillis();
+            predictResult = predictWeightContext.getAmazonMachineLearningClient().predict(predictWeightContext.getPredictRequest());
```
Comment (syedmusamah) [inline src/main/java/com/thredup/callable/PredictWeightCallable.java]: The `total_item_count` variable is missing from the call to AWS: 

screen shot 2017-01-19 at 12 35 01 pm

Comment (syedmusamah) [inline src/main/java/com/thredup/callable/PredictWeightCallable.java]: Also, the model expects `"#{merch_department}_item_count"` for every merch department. We should pass "0" if that merch department does not exist in the current order/cart/items_collection

Comment (florrain) [inline src/main/java/com/thredup/callable/PredictWeightCallable.java]: 👀  the Jetbrains darcula theme ^ 🥇 💯

Comment (mrmdrogers) [inline src/main/java/com/thredup/callable/PredictWeightCallable.java]: 👍

File: src/main/java/com/thredup/constants/MerchandisingDepartment.java (hunk 3 above)
Comment (mrmdrogers) [inline src/main/java/com/thredup/constants/MerchandisingDepartment.java]: 👍

File: src/main/java/com/thredup/datatype/DimensionPredictRequestContext.java (hunk 2 above)
Comment (mrmdrogers) [inline src/main/java/com/thredup/datatype/DimensionPredictRequestContext.java]: 👍

File: src/main/java/com/thredup/configuration/AwsConfiguration.java (hunk 1 above)
Comment (mrmdrogers) [inline src/main/java/com/thredup/configuration/AwsConfiguration.java]: 👍 Good catch

**Pull request merged**